streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTablePrefixValueGetter.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.state.KeyValueIterator;
+
+public interface KTablePrefixValueGetter<K, V> extends KTableValueGetter<K, V> {
+
+    KeyValueIterator<K, V> prefixScan(final K prefix);
```

Comment:
I put this in its own interface for now because it's not immediately clear where it should live. It needs to be part of StateStore, given how all of the various wrapper classes, like the metered, caching and logging ones work. However, I do not want it to be in the ReadOnlyKeyValueStore as it shouldn't be exposed to the outside world.